pkg/testcommon/testcommon.go

```diff
@@ -297,8 +296,13 @@ func ContainerCheck(checkName string, checkState string) (bool, error) {
 // internalExtractionPath is the place in the archive to start extracting
 // sourceURL is the actual URL to download.
 // Returns the extracted path, the tarball path (both possibly cached), and an error value.
-func GetCachedArchive(siteName string, prefixString string, internalExtractionPath string, sourceURL string) (string, string, error) {
-	uniqueName := prefixString + "_" + path.Base(sourceURL)
+func GetCachedArchive(siteName, prefixString, internalExtractionPath, sourceURL string) (string, string, error) {
+	var uniqueName string
+	if strings.Contains(sourceURL, ".tar.") {
+		uniqueName = fmt.Sprintf("%x.tar%s", sha256.Sum256([]byte(sourceURL)), path.Ext(sourceURL))
+	} else {
+		uniqueName = fmt.Sprintf("%x%s", sha256.Sum256([]byte(sourceURL)), path.Ext(sourceURL))
+	}
```

Comment:
Create a unique name based on the source URL only so we have each asset only once in the cache.